nucleus/deployment/common/src/main/java/org/glassfish/deployment/common/InstalledLibrariesResolver.java

```diff
@@ -16,325 +17,298 @@
-                    extName = extName.trim();
+            StringTokenizer extensionListTokenizer = new StringTokenizer(extensionList, " ");
+            while (extensionListTokenizer.hasMoreTokens()) {
+
+                String token = extensionListTokenizer.nextToken().trim();
+                String extensionName = manifest.getMainAttributes().getValue(token + "-" + Attributes.Name.EXTENSION_NAME);
+                if (extensionName != null) {
+                    extensionName = extensionName.trim();
                 }
 
-                String specVersion = manifest.getMainAttributes().
-                        getValue(token + "-" + Attributes.Name.SPECIFICATION_VERSION);
+                String specVersion = manifest.getMainAttributes().getValue(token + "-" + Attributes.Name.SPECIFICATION_VERSION);
 
-                Extension extension = new Extension(extName);
+                Extension extension = new Extension(extensionName);
                 if (specVersion != null) {
                     extension.setSpecVersion(specVersion);
                 }
-                //TODO possible NPE when extension is unspecified
+
+                // TODO possible NPE when extension is unspecified
                 boolean isLibraryInstalled = libraryStore.containsKey(extension);
                 if (isLibraryInstalled) {
-                    String libraryName = libraryStore.get(extension);
-                    libraries.add(libraryName);
-                } else {
-                    if(strict){
-                        throw new MissingResourceException(extName + " not found", extName, null);
-                    }else{
-                        // no action needed
-                    }
+                    libraries.add(libraryStore.get(extension));
+                } else if (strict) {
+                    throw new MissingResourceException(extensionName + " not found", extensionName, null);
                 }
-                if (deplLogger.isLoggable(Level.FINEST)) {
-                    deplLogger.log(Level.FINEST, " is library installed [" + extName + "] " +
-                            "for archive [" + archiveURI + "]: " + isLibraryInstalled);
+
+                if (deplLogger.isLoggable(FINEST)) {
+                    deplLogger.log(FINEST,
+                            " is library installed [" + extensionName + "] " + "for archive [" + archiveURI + "]: " + isLibraryInstalled);
                 }
             }
         }
+
         return libraries;
     }
 
-
     /**
-     * @return a list of libraries included in the archivist
+     * @return a list of libraries included in the archive
      */
-    private static Vector getArchiveLibraries(ReadableArchive archive) {
-
+    private static List<String> getEmbeddedLibraries(ReadableArchive archive) {
         Enumeration<String> entries = archive.entries();
-        if (entries == null)
-            return null;
+        if (entries == null) {
+            emptyList();
```

Comment:
This is unclear for me.
- What happens with that empty list?
- What happens in `while` below with `entries == null` then?

Reply:
Seems to me there is missing the `return` keyword. 
And I had the same mistake yesterday in my code, seems Eclipse editor sometimes does something ... not ideal :-D

Reply:
Thanks so much for noticing, do I see a PR coming? ;)